martin-tile-utils/src/lib.rs

```diff
@@ -248,13 +258,22 @@ pub fn webmercator_to_wgs84(x: f64, y: f64) -> (f64, f64) {
     (lng, lat)
 }
 
+#[must_use]
+pub fn wgs84_to_webmercator(lon: f64, lat: f64) -> (f64, f64) {
+    let x = EARTH_CIRCUMFERENCE / 360.0 * lon;
+    let y = ((90.0 + lat) * PI / 360.0).tan().ln() / (PI / 180.0);
+    let y = EARTH_CIRCUMFERENCE / 360.0 * y;
```

Comment:
i merged the two suggestions here, but as the result, it no longer matches the expected result due to rounding (?).   I suspect that you would need to call `.min(1<<zoom)` somewhere

Reply:
oh, and another suspicious thing - are you sure y should cover 360?  maybe it does, i don't know.  Make sure you read https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames

Reply:
This code is from our private server. TBH I don't remember the source....   

I've adopted some code from [esri](https://github.com/Esri/arcgis-osm-editor/blob/e4b9905c264aa22f8eeb657efd52b12cdebea69a/src/OSMWeb10_1/Utils/WebMercator.cs). Sadly i'm bad at math😔,any help?